### PR TITLE
added cpp/matthewrolls solution, associated items

### PR DESCRIPTION
**Pull request by mrolls13**

### CODE REVIEW

**Comment by LukasB4** [comment]
+1 Good code

**Review by LukasB4 — APPROVED** [review]

**Comment by yabetsebren** [comment]
It didn't work when i run the script locally. Segmentation fault. You may want to look in to that. 

Screenshot 2024-03-19 at 5 18 15 PM

**Comment by mrolls13** [comment]
> It didn't work when i run the script locally. Segmentation fault. You may want to look in to that.
> Screenshot 2024-03-19 at 5 18 15 PM

It still runs in my vm, I might have messed up something when transferring it. I'll look into it. Did you actually enter a number? There is no exception for null input and I get a segmentation fault as well if I just hit enter after prompted for a number.

**Comment by Jake-Ripley** [comment]
Solution doesn't work, segmentation fault
Screenshot 2024-03-19 at 6 21 43 PM

**Comment by mrolls13** [comment]
> Solution doesn't work, segmentation fault Screenshot 2024-03-19 at 6 21 43 PM

Screenshot (946)
 This is using the linux vm for csci476

**Review by Jake-Ripley — APPROVED** [review]
+1 looks good

File: problems/fizzbuzz/cpp/matthewrolls/script.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+echo Enter number
```
Comment (sleddog) [inline problems/fizzbuzz/cpp/matthewrolls/script.sh]: do not ask the number here during the script.  this should be passed as an argument to the shell script:  `./script.sh 16`

File: problems/fizzbuzz/cpp/matthewrolls/script.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+g++ -o fzbz-mr -Wall fzbz-mr.cpp
+./fzbz-mr $num
```
Comment (sleddog) [inline problems/fizzbuzz/cpp/matthewrolls/script.sh]: pass the value using `$1`:   `./fzbz-mr $1`

**Review by sleddog — CHANGES_REQUESTED** [review]
please make the changes requested above

File: problems/fizzbuzz/cpp/matthewrolls/fzbz-mr.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		cout<<out;
+	}else{
+		cout<<"Testing\n";
+		assert("1\n2\nFizz\n"==fb(3));
+		cout<<"First test passed\n";
+		assert("1\n2\nFizz\n4\nBuzz\n"==fb(5));
+		cout<<"Second test passed\n";
+		assert("1\n2\nFizz\n4\nBuzz\nFizz\n7\n8\nFizz\nBuzz\n11\nFizz\n13\n14\nFizzBuzz\n"==fb(15));
+		cout<<"Third test passed\n";
+	}
```
Comment (sleddog) [inline problems/fizzbuzz/cpp/matthewrolls/fzbz-mr.cpp:37]: these tests should be in a separate file

**Comment by sleddog** [comment]
+10 
image

**Review by sleddog — APPROVED** [review]
+1

**Pull request merged**